File: Data Structures & Algorithms/missing-number/submission-4.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        # 0, 1, 3, 4, 5; s = 15, a0 = 0 an = 5 n = 6
        n = len(nums)
        an = n+1
        a0 = 0
        summ = (n/2)*(a0+an)
        return int(summ - sum(nums))
        
        
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()<2){
            return nums.size();
        }
        sort(nums.begin(), nums.end());
        int res = 1;
        int best = res;
        for(int i = 1; i < nums.size(); i++) {
            cout << nums[i];
            int diff = nums[i] - nums[i-1];
            if(diff == 0) {
                continue;
            }
            // -3-2-1 1 ; 2 3 4 6 ; 7 8 9
            else if(diff > 1) {
                cout << diff << " is greater than 1\n";
                if(res >= best) {
                    best = res;
                    res = 1;
                }
                continue;
            }
            res ++;
        }
        if(res > best) {
            best = res;
        }
        return best;
    }
};
